app/src/main/java/org/sagebionetworks/research/mpower/MainFragment.kt

```diff
@@ -53,9 +66,43 @@ import javax.inject.Inject
  * A simple [Fragment] subclass.
  *
  */
-class MainFragment : DaggerFragment() {
+class MainFragment : DaggerFragment(), OnRequestPermissionsResultCallback {
     private val LOGGER = LoggerFactory.getLogger(MainFragment::class.java)
 
+    private val runningQOrLater = Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q
+    private var activityTrackingEnabled = false
+    private val activityTransitions: MutableList<ActivityTransition> by lazy {

```

Comment:
Do these need to live in the MainFragment or can they go elsewhere?

Reply:
for now i am having it live in MainFragment just so it is not started up before user is authenticated.  In future updates, i will move to the EntryActivity and control enabling/disabling tracking via an event so that app can always signal to stop passive gait tracking (e.g. - if user is starting some other activity)

Reply:
Cool.  Can i ask you to add a card to Pivotal so that we don't lose track of it?